ecs-agent/httpclient/httpclient.go

```diff
@@ -44,18 +53,30 @@ type ecsRoundTripper struct {
 	transport          http.RoundTripper
 }
 
-func (client *ecsRoundTripper) userAgent() string {
+// userAgent generates the User-Agent header value for HTTP requests that the agent makes to ECS
+// Ref: https://www.rfc-editor.org/rfc/rfc9110.html#name-user-agent
+func (client *ecsRoundTripper) userAgent(req *http.Request) string {
+	var userAgentValue string
+
 	// Include detailedOSFamily when available
 	if client.detailedOSFamily != "" {
-		return fmt.Sprintf("%s (%s; %s) (+http://aws.amazon.com/ecs/)",
+		userAgentValue = fmt.Sprintf("%s (%s; %s) (+http://aws.amazon.com/ecs/)",
 			client.agentVersion, client.osType, client.detailedOSFamily)
+	} else {
+		userAgentValue = fmt.Sprintf("%s (%s) (+http://aws.amazon.com/ecs/)", client.agentVersion, client.osType)
 	}
-	// Fallback to current behavior when detailedOSFamily is not set
-	return fmt.Sprintf("%s (%s) (+http://aws.amazon.com/ecs/)", client.agentVersion, client.osType)
+
+	// In some cases like queries to TMDS, the ECS agent makes requests to ECS on behalf of the task containers.
+	// Append the container name when available, for CloudTrail visibility
+	if containerName, ok := req.Context().Value(ContainerNameKey).(string); ok {
+		userAgentValue = fmt.Sprintf("%s container/%s", userAgentValue, containerName)
```

Comment:
Why are we not using a new header for context instead of tacking it on to user agent?

Reply:
No particular reason. Do you have a better place to put this new information?